### PR TITLE
prevent window exit if txStatus is pending

### LINKED ISSUE #172: Pop up a confirmation message when user wants to close the browser
... while the transaction is in progress, as it might causes unexpected issues, e.g deposit doesn't go through.

Use [beforeunload](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event) event

### PR DESCRIPTION
**Pull request by aidan-starke**
## Description

- closes #172

### CODE REVIEW

File: components/BridgeProgress.tsx [hunk 1]
```diff
@@ -20,6 +20,17 @@ const BridgeProgress: VFC<IntrinsicElements["div"] & BridgeProgressProps> = (
 	const dismissible =
 		txStatus?.status === "Success" || txStatus?.status === "Failure";
 
+	useEffect(() => {
+		const beforeUnload = (event: BeforeUnloadEvent) => {
+			event.preventDefault();
+			if (txStatus?.status === "Pending") return (event.returnValue = "");
```
Comment (ken-centrality) [inline components/BridgeProgress.tsx:26]: set the `returnValue` to be "There is pending transaction. Are you sure you want to leave?"

That should pop up the message as above when you try to close the browser

**Review by ken-centrality — CHANGES_REQUESTED** [review]
Can you also add this to Pool and Swap too?

File: components/BridgeProgress.tsx (hunk 1 above)
Comment (aidan-starke) [reply components/BridgeProgress.tsx]: done, but I still just get "Changes that you made may not be saved." in popup

Comment (aidan-starke) [reply components/BridgeProgress.tsx]: Screen Shot 2022-04-22 at 7 51 54 AM
https://stackoverflow.com/questions/38879742/is-it-possible-to-display-a-custom-message-in-the-beforeunload-popup

File: hooks/useBeforeUnload.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { useEffect } from "react";
+
+export default function useBeforeUnload(txStatus) {
```
Comment (ken-centrality) [inline hooks/useBeforeUnload.ts:3]: Good idea to put it in a custom hook 👏 
Can you add type to `txStatus`

File: components/BridgeProgress.tsx (hunk 1 above)
Comment (ken-centrality) [reply components/BridgeProgress.tsx]: sweet

**Review by ken-centrality — CHANGES_REQUESTED** [review]
one small change, then happy for you to merge

**Review by eupston — APPROVED** [review]
lgtm

**Review by ken-centrality — APPROVED** [review]

**Pull request merged**